### PR TITLE
UIOR-348 Store/Retrieve order templates in new order-templates API

### PR DESCRIPTION
**Pull request by Feboll**
## Purpose

https://issues.folio.org/browse/UIOR-348

## Approach

Change api for create order templates

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/settings/OrderTemplates/OrderTemplateView/OrderTemplateView.js [hunk 1]
```diff
@@ -159,21 +159,20 @@ class OrderTemplateView extends Component {
       contributorNameTypes,
     } = this.props;
     const { sections, showConfirmDelete } = this.state;
-    const orderTemplate = get(template, 'orderTemplate', {});
```
Comment (aliaksei-chumakou) [inline src/settings/OrderTemplates/OrderTemplateView/OrderTemplateView.js]: was it necessary to change `orderTemplate` name to `template`? Isn't it easier to change one line instead - `template: orderTemplate,`?

File: src/settings/OrderTemplates/OrderTemplatesEditor/OrderTemplatesEditorContainer.js [hunk 2]
```diff
@@ -107,12 +88,12 @@ class OrderTemplatesEditorContainer extends Component {
...
     const materialTypes = getMaterialTypesForSelect(resources);
-    const orderTemplate = getOrderTemplatesList(get(resources, 'orderTemplate.records', []));
+    const orderTemplate = get(resources, ['orderTemplate', 'records', 0], {});
     const id = get(match, ['params', 'id']);
     const template = id
-      ? get(orderTemplate, 0, {})
+      ? { orderTemplate }
       : { orderTemplate: INITIAL_VALUES };
-    const title = get(template, 'title') || <FormattedMessage id="ui-orders.settings.orderTemplates.editor.titleCreate" />;
+    const title = get(template, ['orderTemplate', 'templateName']) || <FormattedMessage id="ui-orders.settings.orderTemplates.editor.titleCreate" />;
```
Comment (aliaksei-chumakou) [inline src/settings/OrderTemplates/OrderTemplatesEditor/OrderTemplatesEditorContainer.js]: are you sure about `'orderTemplate'` here?

File: src/components/Utils/resources.js [hunk 3]
```diff
@@ -235,18 +236,13 @@ export const SUFFIXES_SETTING = {
...
   ...BASE_RESOURCE,
-  path: CONFIG_API,
-  records: 'configs',
-  GET: {
-    params: {
-      query: `(module=${MODULE_ORDERS} and configName=${CONFIG_ORDER_TEMPLATES})`,
-    },
-  },
+  path: ORDER_TEMPLATES_API,
+  records: 'orderTemplates',
```
Comment (AndreiShumski2) [inline src/components/Utils/resources.js]: `{ params: query: 'cql.allRecords=1 sortby  templateName' }` for default alphabetic order

Comment (Feboll) [reply src/components/Utils/resources.js]: updated

File: src/settings/OrderTemplates/OrderTemplatesEditor/OrderTemplatesEditorContainer.js (hunk 2 above)
Comment (Feboll) [reply src/settings/OrderTemplates/OrderTemplatesEditor/OrderTemplatesEditorContainer.js]: changed

File: src/settings/OrderTemplates/OrderTemplateView/OrderTemplateView.js (hunk 1 above)
Comment (Feboll) [reply src/settings/OrderTemplates/OrderTemplateView/OrderTemplateView.js]: changed

**Pull request merged**